src/Renderer/UI/UpdateHelpers.fs

```diff
@@ -412,25 +431,37 @@ let processContextMenuClick
         |> withNoCmd
 
     | DBComp sym, "Properties" | DBCustomComp(sym, _), "Properties" ->
-       model
-       |> set selectedComponent_ (Some sym.Component)
-       |> set (sheet_ >-> SheetT.selectedWires_) []
-       |> set (sheet_ >-> SheetT.selectedComponents_) [sym.Id]
-       |> set rightPaneTabVisible_ Properties
-       |> withMsg (Msg.Sheet (SheetT.Msg.Wire (BusWireT.Msg.Symbol (SymbolT.SelectSymbols [sym.Id]))))
+         model
+        |> set selectedComponent_ (Some sym.Component)
+        |> set (sheet_ >-> SheetT.selectedWires_) []
+        |> set (sheet_ >-> SheetT.selectedComponents_) [sym.Id]
+        |> set rightPaneTabVisible_ Properties
+        |> withMsg (Msg.Sheet (SheetT.Msg.Wire (BusWireT.Msg.Symbol (SymbolT.SelectSymbols [sym.Id]))))
 
     | DBComp sym, "Rotate Clockwise (Ctrl-Right)" ->
         rotateDispatch SymbolT.RotateClockwise
         model
         |> set (sheet_ >-> SheetT.selectedComponents_) [sym.Id]
         |> withNoCmd
+    
+    | DBScalingBox selectedcomps, "Rotate Clockwise (Ctrl-Right)"->
+        rotateDispatch SymbolT.RotateClockwise
+        model 
+        // |> set (sheet_ >-> SheetT.Action_) SheetT.Idle
+        |> withMsg (Msg.Sheet (SheetT.Msg.Wire (BusWireT.Msg.UpdateConnectedWires selectedcomps)))
 
     | DBComp sym, "Rotate AntiClockwise (Ctrl-Left)" ->
         rotateDispatch SymbolT.RotateAntiClockwise
         model
         |> set (sheet_ >-> SheetT.selectedComponents_) [sym.Id]
         |> withNoCmd
 
+    | DBScalingBox selectedcomps, "Rotate AntiClockwise (Ctrl-Left)"->
+        rotateDispatch SymbolT.RotateAntiClockwise
+        model 
+        // |> set (sheet_ >-> SheetT.Action_) SheetT.Idle
+        |> withMsg (Msg.Sheet (SheetT.Msg.Wire (BusWireT.Msg.UpdateConnectedWires selectedcomps)))
```

Comment:
Refactor (it is used serval times). Create function `withWireMsg` at top of update and use where needed